modules/seqexec/web/client/src/main/scala/seqexec/web/client/components/sequence/steps/StepsTable.scala

```diff
@@ -463,13 +467,27 @@ object StepsTable extends Columns {
 
   type Props = StepsTable
 
+  final case class StepSummary(id: StepId, status: StepState, breakpoint: Boolean)
+  object StepSummary {
+    def fromStep(step: Step): StepSummary = StepSummary(step.id, step.status, step.breakpoint)
+
+    implicit val stepSummaryEq: Eq[StepSummary] = Eq.fromUniversalEquals
+  }
+
   @Lenses
   final case class State(
-    tableState:      TableState[TableColumn],
-    breakpointHover: Option[Int],
-    selected:        Option[StepId],
-    scrollCount:     Int,
-    scrollBarWidth:  Double
+    tableState:               TableState[TableColumn],
+    breakpointHover:          Option[Int],
+    selected:                 Option[StepId],
+    scrollCount:              Int,
+    scrollBarWidth:           Double,
+    prevStepSummaries:        List[StepSummary],
```

Comment:
Ouch, it really starts to get complicated keeping the previous props in state

Reply:
Yes, this was the worst case. As mentioned before, I tried to port the existing logic. I don't know if there's another approach that will simplify this. I suspect there isn't without restructuring a lot of things.